lib/id3c/cli/command/etl/__init__.py

```diff
@@ -78,23 +78,64 @@ def upsert_individual(db: DatabaseSession, identifier: str, sex: str = None, det
     data = {
         "identifier": identifier,
         "sex": sex,
-        "details": Json(details),
+        "details": Json(details) if details else None,
     }
 
-    individual = db.fetch_row("""
-        insert into warehouse.individual (identifier, sex, details)
-            values (%(identifier)s, %(sex)s, %(details)s)
+    with db.cursor() as cursor:
+        cursor.execute("""
+            select individual_id as id,
+                identifier,
+                (
+                    individual.sex != %(sex)s
+                    or
+                    (coalesce(individual.details,'{}'::jsonb) != (coalesce(individual.details, '{}'::jsonb) || coalesce(%(details)s, '{}'::jsonb)))
```

Comment:
Curious about this line, in what circumstances would `(coalesce(individual.details,'{}'::jsonb) != (coalesce(individual.details, '{}'::jsonb)` these two things not be equal to each other? Does it have something to do with the null values you noted in your comment above?

Reply:
The two values being compared there are:

`coalesce(individual.details,'{}'::jsonb)` 
and
`(coalesce(individual.details, '{}'::jsonb) || coalesce(%(details)s, '{}'::jsonb))`

The latter is the merged value that would be used if an update is performed. And yes, the coalesces are there to make sure it handles nulls correctly. The current version is resulting in jsonb arrays where a null is appended each time there's an update but this change should prevent that.

Reply:
Ah got it thanks, I was misreading the statement.